### DISCUSSION TITLE
Custom coloring for bed tracks

### DISCUSSION THREAD

**Discussion by barslmn** [GMOD/jbrowse-components#2561] [Q&A] [answered]
Hey,

Thanks for the great viz tool. I have a question regarding coloring. How can I color bed tracks based on an attribute?  To make it more clear I have this example bed file:

chr1 300 500 SampleA 1
chr2 500 700 SampleB 3
chr3 700 900 SampleC 2 LOH

I want to color tracks red if the fifth color is 1 and green if it is 3. And some other color if the sixth column is LOH.
I had no problems with adding an itemRGB column and displaying it with colors in the UCSC browser however jbrowse doesn't seem to care about that either. 
Is there are a way to color the bed tracks or is there a supported format that is used to color tracks?
Best,
B

**Comment by cmdcolin** [comment, marked as answer]
Hi there

Currently we don't support itemRgb field in the BED, it would be a good feature and I attempted to get it added but was a little too hacky :)

There is a ticket for it here https://github.com/GMOD/jbrowse-components/issues/1734

Currently you could use a customization to modify the color using a "jexl" callback

I made this work with your data by adding this BED file as a tabix BED track (bgzip yourfile.bed; tabix yourfile.bed.gz) then then added it to the config `jbrowse add-track yourfile.bed.gz --load copy` and then updating the config by hand to look like this

```
{
      "type": "FeatureTrack",
      "trackId": "test.bed",
      "name": "test.bed",
      "assemblyNames": ["hg19"],
      "adapter": {
        "type": "BedTabixAdapter",
        "bedGzLocation": {
          "uri": "test.bed.gz",
          "locationType": "UriLocation"
        },
        "index": {
          "location": {
            "uri": "test.bed.gz.tbi",
            "locationType": "UriLocation"
          },
          "indexType": "TBI"
        }
      },
      "displays": [
        {
          "type": "LinearBasicDisplay",
          "displayId": "anythingcangohere",
          "renderer": {
            "type": "SvgFeatureRenderer",
            "color1": "jexl:get(feature,'field5')=='LOH'?'blue':'red'"
          }
        }
      ]
    }
```

looks like this (modified bed to be all features on chr1 too)

![localhost_3000__config=test_data%2Fconfig json session=local-GOEFyqnzo](https://user-images.githubusercontent.com/6511937/145632481-d9f1585c-b541-4f9c-a6fc-67f64be83edb.png)

hope that helps. our config system may look kinda verbose too but let me know if there is anything i can help clarify

**Reply by barslmn** [reply]
Thanks a lot, this solves the issue neatly. However, editing the config by hand doesn't seem to be the best approach since jBrowse2 doesn't seem to apply the changes when the config file is saved by another text editor. So I have to close and reopen the session.  I noticed the color1 field in the track settings on the frontend also accepts jexl callbacks, so just editing there is more practical since changes are applied immediately.  

![Capture](https://user-images.githubusercontent.com/43095157/145710522-8d2eafa0-fbe0-4dbd-801f-381b08803003.PNG)

**Reply by cmdcolin** [reply]
awesome, glad you got the config editor to work as well :)!

**Reply by cmdcolin** [reply]
note that you could also maybe simplify your code using something like this

'rgb('+get(feature, 'field8')+')'
